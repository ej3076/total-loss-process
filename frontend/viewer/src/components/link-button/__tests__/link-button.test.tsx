import { mount } from 'enzyme';
import React from 'react';
import { MemoryRouter, Router, RouterProps } from 'react-router';
import { create } from 'react-test-renderer';

import LinkButton from '../';

describe('<LinkButton />', () => {
  it('should render correctly', () => {
    const tree = create(
      <MemoryRouter>
        <LinkButton to="/foo" />
      </MemoryRouter>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render with additional props', () => {
    const tree = create(
      <MemoryRouter>
        <LinkButton
          to="/foo"
          // blueprint prop
          intent="primary"
          // html props
          aria-label="hello"
          data-my-custom-data-attribute="foobar"
        />
      </MemoryRouter>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should navigate on click', () => {
    const wrapper = mount(
      <MemoryRouter>
        <LinkButton to="/foo" />
      </MemoryRouter>,
    );
    const button = wrapper.find(LinkButton);
    const { history } = wrapper.find<RouterProps>(Router).props();

    expect(history.length).toBe(1);
    expect(history.location.pathname).toBe('/');

    button.simulate('click');

    expect(history.length).toBe(2);
    expect(history.location.pathname).toBe('/foo');
  });
});
