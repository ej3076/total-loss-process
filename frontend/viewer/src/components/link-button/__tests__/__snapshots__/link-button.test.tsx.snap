// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkButton /> should render correctly 1`] = `
<a
  className="bp3-button"
  onClick={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
/>
`;

exports[`<LinkButton /> should render with additional props 1`] = `
<a
  aria-label="hello"
  className="bp3-button bp3-intent-primary"
  data-my-custom-data-attribute="foobar"
  onClick={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
/>
`;
